app/core/Encryptor/Encryptor.ts

```diff
@@ -222,10 +224,11 @@ class Encryptor implements WithKeyEncryptor<EncryptionKey, Json> {
     //   * use a different number of iterations for the KDF
     const key = await this.keyFromPassword(
       password,
-      payload.salt,
+      salt,
       false,
-      payload.keyMetadata ?? LEGACY_DERIVATION_OPTIONS,
-      payload.lib,
+      // If the keyMetadata is not present, we can assume the key was derived using the legacy options
+      keyMetadata || LEGACY_DERIVATION_OPTIONS,
```

Comment:
Unless `keyMetadata` could be something falsy?

```suggestion
      keyMetadata ?? LEGACY_DERIVATION_OPTIONS,
```